src/controllers/interaction.ts

```diff
@@ -40,6 +40,8 @@ export const interactionStartControllerFactory =
       req.session.mustReturnOneOrganizationInPayload =
         mustReturnOneOrganizationInPayload(scope);
       req.session.twoFactorsAuthRequested = twoFactorsAuthRequested(prompt);
+      req.session.spName =
+        typeof params["sp_name"] === "string" ? params["sp_name"] : undefined;
```

Comment:
todo : should be set line 35 like other params ; also update the type OIDCContextParams.

Reply:
suggestion : use logical or for this
```suggestion
        sp_name: sp_name || undefined
```